q2_chemistree/_collate_fingerprint.py

```diff
@@ -48,15 +56,25 @@ def collate_fingerprint(csi_result: CSIDirFmt) -> biom.Table:
                 with open(os.path.join(fidpath, 'fingerprints', fname)) as f:
                     fp = f.read().strip().split('\n')
                 molfp[fid] = fp
-
     fingerids = pd.DataFrame.from_dict(molfp, orient='index')
     if fingerids.shape == (0, 0):
         raise ValueError('Fingerprint file is empty!')
+    substructrs = pd.read_table(os.path.join(csi_result, 'fingerprints.csv'),
+                                index_col='absoluteIndex')
     fingerids.index.name = '#featureID'
+    fingerids.columns = substructrs.index
+    if qc_properties:
+        fingerids = select_pubchem(fingerids)
     npfid = np.asarray(fingerids)
-
     # biom requires that ids be strings
     fptable = biom.table.Table(data=npfid,
                                observation_ids=fingerids.index.astype(str),
                                sample_ids=fingerids.columns.astype(str))
     return fptable
+
+
+def select_pubchem(fingerids):
+    properties = pd.read_table(os.path.join(data, 'molecular_properties.csv'),
```

Comment:
Looks like this function might be a bit of overkill, specially when you consider that you also have to load the table, maybe it would be better to put this function inline with the rest of the code i.e. as part of the `if qc_properties` branch.